src/main/java/net/minecraftforge/client/CloudRenderer.java

```diff
@@ -0,0 +1,495 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client;
+
+import java.nio.ByteBuffer;
+
+import org.lwjgl.opengl.GL11;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.EntityRenderer;
+import net.minecraft.client.renderer.GLAllocation;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.OpenGlHelper;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.texture.DynamicTexture;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexBuffer;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.client.resources.IReloadableResourceManager;
+import net.minecraft.client.resources.IResourceManager;
+import net.minecraft.client.resources.IResourceManagerReloadListener;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.common.ForgeModContainer;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent.ClientTickEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent.Phase;
+
+public class CloudRenderer implements IResourceManagerReloadListener
+{
+    // Shared constants.
+    private static final float PX_SIZE = 1 / 256F;
+
+    // Building constants.
+    private static final VertexFormat FORMAT = DefaultVertexFormats.POSITION_TEX_COLOR;
+    private static final int TOP_SECTIONS = 12;	// Number of slices a top face will span.
+    private static final int HEIGHT = 4;
+    private static final float INSET = 0.001F;
+    private static final float ALPHA = 0.8F;
+
+    // Debug
+    private static final boolean WIREFRAME = false;
+
+    // Instance fields
+    private final Minecraft mc = Minecraft.getMinecraft();
+    private final ResourceLocation texture = new ResourceLocation("textures/environment/clouds.png");
+
+    private int displayList = -1;
+    private VertexBuffer vbo;
+    private int cloudMode = -1;
+    private int renderDistance = -1;
+
+    private DynamicTexture COLOR_TEX = null;
+
+    private int texW;
+    private int texH;
+
+    public CloudRenderer()
+    {
+        // Resource manager should always be reloadable.
+        ((IReloadableResourceManager) mc.getResourceManager()).registerReloadListener(this);
+
+        MinecraftForge.EVENT_BUS.register(this);
```

Comment:
I tried again with a new workspace but this line still causes an error log.
There is no active mod container when this is called.

```
[main/ERROR] [FML]: Unable to determine registrant mod for net.minecraftforge.client.CloudRenderer@72f05bd9. This is a critical error and should be impossible
java.lang.Throwable: null
	at net.minecraftforge.fml.common.eventhandler.EventBus.register(EventBus.java:74) [EventBus.class:?]
	at net.minecraftforge.client.CloudRenderer.<init>(CloudRenderer.java:84) [CloudRenderer.class:?]
	at net.minecraftforge.fml.client.FMLClientHandler.onInitializationComplete(FMLClientHandler.java:463) [FMLClientHandler.class:?]
	at net.minecraft.client.Minecraft.init(Minecraft.java:552) [Minecraft.class:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:377) [Minecraft.class:?]
	at net.minecraft.client.main.Main.main(Main.java:118) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
	at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:97) [Forge_main/:?]
	at GradleStart.main(GradleStart.java:26) [Forge_main/:?]
```

I think it has to be called from somewhere when FML is the active mod container.
Make sure you are testing with your branch merged into the latest for the 1.12.x branch.